sympy/simplify/combsimp.py

```diff
@@ -66,9 +64,14 @@ def combsimp(expr):
 
     # as a rule of thumb, if the expression contained gammas initially, it
     # probably makes sense to retain them
-    as_gamma = expr.has(gamma)
     as_factorial = expr.has(factorial)
     as_binomial = expr.has(binomial)
+    as_gamma = expr.has(gamma)
+
+    if expr.has(gamma):
+        if not expr.has(factorial):
```

Comment:
Maybe all expressions involving gamma should be dispatched to gammasimp regardless of whether they also have factorial (or binomial) or not, at least before the simplification of purely combinatorial expressions has been worked out. Then it would not be necessary to consider gamma in this function.